wagtail/admin/views/pages/preview.py

```diff
@@ -43,6 +46,16 @@ def get_form(self, query_dict):
                 for_user=self.request.user,
             )
 
+        query_dict = query_dict.copy()
+
+        if not query_dict.get("title"):
+            query_dict["title"] = gettext("Placeholder title")
+
+        # Generate a random slug if one is not provided, use UUID to ensure
+        # uniqueness without needing to hit the database
+        if not query_dict.get("slug"):
+            query_dict["slug"] = str(uuid.uuid4())
```

Comment:
I thought it'd be nice to allow these to be blank so that the preview can be displayed immediately when creating a page (if no other validation errors prevent it).

I used uuid just to ensure uniqueness. We can also use something else such as the `session_key`, but it's less robust because if for some reason there's a sibling page with the same value (e.g. `wagtail-preview-myapp-mymodel-123`) then we'd still get a validation error. It's a very unlikely edge case though, so if we'd rather use it instead of generating a UUID, I'd be okay with it too.

Reply:
This is a really nice improvement! Avoiding the initial "can't show preview" message is a small thing that will improve the user experience a lot.

I see no problem with using a UUID - the slug is unlikely to surface in any user-visible way here, so it's pretty much arbitrary what we use here.